### ISSUE TITLE
VirtualBoxOVA does not work for more than 64gb baseImageSize

### ISSUE THREAD

**Issue by dfordivam** [NixOS/nixpkgs#59867]
## Issue description
The standard way to create a nixos virtual box .ova files (`config.system.build.virtualBoxOVA`) does not work if the `baseImageSize` is more than 64gb.

### Steps to reproduce
Specify baseImageSize more than 64gb 

    virtualbox = {
        baseImageSize = 65 * 1024; # in MiB
    }

## Technical details
The VM creation hangs with this output

    [ 3.565686] This architecture does not have kernel memory protection. 
    [ 3.595800] EXT4-fs (vda1): mounted filesystem with ordered data mode. Opts: 
    [ 6.028488] random: crng init done

This is the `strace` from `cptofs` which does 100% cpu usage and does not finish.

    timer_settime(0, 0, {it_interval={tv_sec=0, tv_nsec=0}, it_value={tv_sec=0, tv_nsec=9839123}}, NULL) = 0
    timer_settime(0, 0, {it_interval={tv_sec=0, tv_nsec=0}, it_value={tv_sec=0, tv_nsec=9764387}}, NULL) = 0

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the
results.
 - system: `"x86_64-linux"`
 - host os: `Linux 4.14.104, NixOS, 19.09pre171786.34aa254f9eb (Loris)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.2`
 - channels(root): `"nixos-19.09pre171786.34aa254f9eb"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

**Comment by matthewbauer** [comment]
This was encountered in #49317 as well.

**Comment by dfordivam** [comment]
@matthewbauer I have this workaround #60246 to add an extra disk.

**Comment by matthewbauer** [comment]
Does the extra disk not hit the 64gb limit?

**Comment by dfordivam** [comment]
No, the limit is only because of the bug in `cptofs`, and I do not use this tool. The extra disk is empty with only a filesystem.

**Comment by flokli** [comment]
@dfordivam could you open a bug upstream about cptofs being broken with images > 64GB? I doubt that's a nix-specific thing…

**Comment by ambrop72** [comment]
Why is `cptofs` being used here (specifically in `make-disk-image`)? It's anyway running in `runInLinuxVM` so couldn't we just mount the filesystem and `cp`?

**Comment by ambrop72** [comment]
Oh I see `cptofs` is part of `preVM` which is done before the VM is started. But couldn't it be avoided by sharing $root via 9P and copying the contents inside the VM?

**Comment by matthewbauer** [comment]
Maybe, but cptofs shouldn’t have a 64G limit, right?

**Comment by flokli** [comment]
cptofs definitely shouldn't just burn CPU when working on >64G volumes, so we definitely should open a bug about that upstream.

**Comment by evanjs** [comment]
This is definitely important to me.
Trying to encode sensible defaults into virtual appliances we plan on provisioning, and would love to see this fixed vs implementing _more_ hacky workarounds than we already have regarding ESXi compatibility...

**Comment by evanjs** [comment]
Still important though I’m not sure it makes sense to have this issue open if we’re just waiting on https://github.com/lkl/linux/issues/466.

**Comment by evanjs** [comment]
Still quite important, but blocked on upstream
